src/core/class/GroupIdentifiableUnique.h

```diff
@@ -129,29 +129,42 @@ template<typename T>
 typename GroupIdentifiableUnique<T>::iterator
 GroupIdentifiableUnique<T>::add(const std::unique_ptr<T>& elem)
 {
+    auto iteratorFound = items_.find(elem);
+    if (iteratorFound != items_.end()) {
+        items_.erase(iteratorFound);
+    }
+
     std::unique_ptr<T> res(dynamic_cast<T*>(elem->clone().release()));
     auto it = items_.insert(std::move(res));
     
-    if (it.second) {
-        return it.first;
-    }
-    else {
-        throw std::logic_error("Group: Duplicated Id " + elem->getId().toStr());
+    if (!it.second) {
```

Comment:
I think this cannot happen if we are erasing it just before.

Reply:
Well, looking at it in an atomic way, `insert` can still return a flag in its second element or the pair. Prefer to have that case. Almost nothing is completely impossible-case in programming.

[Set#Insert in CPP Reference](https://en.cppreference.com/w/cpp/container/set/insert)